src/client.ts

```diff
@@ -69,14 +69,20 @@ export class Client<
     const _config = getRuntimeConfig(config) as ClientConfig;
     this.config = _config;
 
-    window.addEventListener('load', () => {
-      if (!this.config.instanceUrl.includes(window.location.origin)) {
-        this.initFrameConduit();
-      }
-    });
+    if (document.readyState === 'complete') {
+      this.initFrameConduit();
+    } else {
+      document.addEventListener('readystatechange', () => {
+        if (document.readyState === 'complete') {
+          this.initFrameConduit();
+        }
+      });
+    }
   }
 
   initFrameConduit() {
+    if (this.config.frameWindow || this.config.instanceUrl.includes(window?.location?.origin)) return;
```

Comment:
just to confirm, this is saying if I have initialized my iframe already or if I "am" the iframe then return early, otherwise run through iframe initialization?

Reply:
thats correct, if we have initialized the iframe already or if we are on a connect origin in which case we don't need the iframe, then return early, otherwise attempt to initialize the iframe.